src/CommunityToolkit/Xamarin.CommunityToolkit/Views/ViewSwitcher/TextSwitcherRenderer.android.cs

```diff
@@ -0,0 +1,558 @@
+		readonly Stack<FormsTextView> childrenStak = new();
+
+		readonly MotionEventHelper motionEventHelper = new();
+		SpannableString spannableString;
+		bool hasLayoutOccurred;
+		bool wasFormatted;
+
+		public TextSwitcherRenderer(Context context)
+			: base(context)
+		{
+			childrenStak.Push(children[0] = new FormsTextView(Context));
+			childrenStak.Push(children[1] = new FormsTextView(Context));
+
+			SetFactory(this);
+			labelTextColorDefault = children[0].TextColors;
+			visualElementRenderer = new VisualElementRenderer(this);
+			BackgroundManager.Init(this);
+		}
+
+		public event EventHandler<VisualElementChangedEventArgs> ElementChanged;
+
+		public event EventHandler<PropertyChangedEventArgs> ElementPropertyChanged;
+
+		VisualElement IVisualElementRenderer.Element => Element;
+
+		VisualElementTracker IVisualElementRenderer.Tracker => visualElementTracker;
+
+		AView IVisualElementRenderer.View => this;
+
+		AView ITabStop.TabStop => this;
+
+		ViewGroup IVisualElementRenderer.ViewGroup => null;
+
+		protected TextSwitcher Element
+		{
+			get => element;
+			set
+			{
+				if (element == value)
+					return;
+
+				var oldElement = element;
+				element = value;
+				OnElementChanged(new ElementChangedEventArgs<Label>(oldElement, element));
+			}
+		}
+
+		protected ATextSwitcher Control => this;
+
+		string? Hint
+		{
+			get => children[0].Hint;
+			set => children.ForEach(c => c.Hint = value);
+		}
+
+		SizeRequest IVisualElementRenderer.GetDesiredSize(int widthConstraint, int heightConstraint)
+		{
+			if (disposed)
+			{
+				return default;
+			}
+
+			if (lastSizeRequest.HasValue)
+			{
+				// if we are measuring the same thing, no need to waste the time
+				var canRecycleLast = widthConstraint == lastConstraintWidth && heightConstraint == lastConstraintHeight;
+
+				if (!canRecycleLast)
+				{
+					// if the last time we measured the returned size was all around smaller than the passed constraint
+					// and the constraint is bigger than the last size request, we can assume the newly measured size request
+					// will not change either.
+					var lastConstraintWidthSize = MeasureSpecFactory.GetSize(lastConstraintWidth);
+					var lastConstraintHeightSize = MeasureSpecFactory.GetSize(lastConstraintHeight);
+
+					var currentConstraintWidthSize = MeasureSpecFactory.GetSize(widthConstraint);
+					var currentConstraintHeightSize = MeasureSpecFactory.GetSize(heightConstraint);
+
+					var lastWasSmallerThanConstraints = lastSizeRequest.Value.Request.Width < lastConstraintWidthSize && lastSizeRequest.Value.Request.Height < lastConstraintHeightSize;
+
+					var currentConstraintsBiggerThanLastRequest = currentConstraintWidthSize >= lastSizeRequest.Value.Request.Width && currentConstraintHeightSize >= lastSizeRequest.Value.Request.Height;
+
+					canRecycleLast = lastWasSmallerThanConstraints && currentConstraintsBiggerThanLastRequest;
+				}
+
+				if (canRecycleLast)
```

Comment:
if we me move this if before `if (!canRecycleLast)` we don't need the `if (!canRecycleLast)`

Reply:
Nope, we can't do it

Also, I am not a fancy to change Forms code that works :)

![image](https://user-images.githubusercontent.com/10124814/122834389-cb500300-d2ee-11eb-988d-a8482917f3f0.png)